### ISSUE TITLE
Add support for mutable branches (not recommended)

### ISSUE THREAD

**Issue by nedtwigg** [diffplug/blowdryer#24]
Blowdryer currently supports GitHub, GitLab, and soon Bitbucket, in the following modes: `COMMIT`, `TAG`, and optionally `TREE`. The key property these modes share is that they are immutable. Blowdryer can compute the URL for any of these resources without making a network call, and once it has downloaded that URL, Blowdryer will never need to request it again. Immutable builds are a good practice, they let Blowdryer support fully offline builds, and it's a performance improvement as well.

Rather than basing your build on links to immutable scripts, you *could* base them instead on mutable scripts, e.g.

```
blowdryerSetup {
  github('acme/blowdryer-acme', 'branch', 'master')
```

The advantage of this would be that as the scripts in `master` improved, all of the builds which rely on them will also improve. But the downside is that your builds will no longer be reproducible. If you check out an old commit, the scripts at `blowdryer-acme` branch `master` might have changed, and so the build results will change. This will make it very difficult to track down regressions, because you can't tell if the changes came from the repo you're looking at, or from the repo you're pointing to from blowdryer.

`build = f(repo)` is easy to work with. Pointing to a mutable branch changes that to `build = f(repo, current_time)`

### But I want to do it anyway

I am pretty sure this is a bad idea, but if you want to do it, I am happy to merge a PR which adds support for mutable branches. If you make such a PR, it should be implemented roughly along these lines:

```java
public class BlowdryerSetup {
  ..
  public enum GitAnchorType { TAG, COMMIT, TREE } // no new branch mode

  public class GitHub {
      String anchor;
      GitAnchorType anchorType;
      ...
      public GitHub branchBuildNotReproducible(String branch) {
        anchorType = GitAnchorType.COMMIT;
        anchor = networkRequestLookupCommitSha(branch);
        return this;
      }
    }
```

And the usage would be

```gradle
blowdryerSetup {
  github('acme/blowdryer-acme', 'TAG', 'UNUSED').branchBuildNotReproducible('master')
```

Basically, if you want to bring mutability into blowdryer, it should all be resolved within the `blowdryerSetup` block, so that the mutability doesn't ripple throughout the system.
